lib/adapters/xhr.js

```diff
@@ -122,11 +122,10 @@ module.exports = function xhrAdapter(config) {
       });
     }
 
-    // Add withCredentials to request if needed
-    if (config.withCredentials) {
-      request.withCredentials = true;
+    // Add withCredentials to request if needed. Explicit to set a true or false
+    if (typeof config.withCredentials !== 'undefined') {
```

Comment:
please use the `isUndefined` function of utils.
https://github.com/axios/axios/blob/841466416b6851666955113a60ae46830a27003f/lib/utils.js#L84-L86